tests/attr/ignore.c

```diff
@@ -31,14 +31,75 @@ static void assert_is_ignored_(
 #define assert_is_ignored(expected, filepath) \
 	assert_is_ignored_(expected, filepath, __FILE__, __LINE__)
 
-void test_attr_ignore__honor_temporary_rules(void)
+static void setup_ignore_config(git_config *cfg)
 {
-	cl_git_rewritefile("attr/.gitignore", "/NewFolder\n/NewFolder/NewFolder");
+	git_buf file = GIT_BUF_INIT, patterns = GIT_BUF_INIT;
+	git_config_iterator *it;
+	git_config_entry *entry;
+	int error;
+
+	cl_git_pass(git_config_iterator_new(&it, cfg));
```

Comment:
What you want here is not an iterator over all values, but to look up `assert.not-ignored` and `assert.ignored` as multivars.

Same with `gitignore.rule`.